src/web-components/KSLHighlightElement.ts

```diff
@@ -166,36 +156,68 @@ export class KSLHighlightElement extends KSLCustomElement {
   }
 
   public connectedCallback(): void {
+    super.connectedCallback();
+
     this.editButtonRef.addEventListener('click', this.handleEditButtonClick);
     this.removeButtonRef.addEventListener('click', this.handleRemoveButtonClick);
   }
 
   public disconnectedCallback(): void {
+    super.connectedCallback();
+
     this.editButtonRef.addEventListener('click', this.handleEditButtonClick);
     this.removeButtonRef.addEventListener('click', this.handleRemoveButtonClick);
     this.unregisterTargetNodeListeners();
   }
 
+  public attributeChangedCallback(attributeName: string, _oldValue: string | null, newValue: string | null): void {
+    if (attributeName === 'deleted') {
```

Comment:
wouldn't it be better to have these attributes as some constants? I see them there and there